presto-rcfile/src/main/java/io/prestosql/rcfile/text/TimestampEncoding.java

```diff
@@ -42,33 +40,24 @@
 public class TimestampEncoding
         implements TextColumnEncoding
 {
-    private static final DateTimeFormatter HIVE_TIMESTAMP_PARSER;
-
-    static {
-        @SuppressWarnings("SpellCheckingInspection")
-        DateTimeParser[] timestampWithoutTimeZoneParser = {
-                DateTimeFormat.forPattern("yyyy-M-d").getParser(),
-                DateTimeFormat.forPattern("yyyy-M-d H:m").getParser(),
-                DateTimeFormat.forPattern("yyyy-M-d H:m:s").getParser(),
-                DateTimeFormat.forPattern("yyyy-M-d H:m:s.SSS").getParser(),
-                DateTimeFormat.forPattern("yyyy-M-d H:m:s.SSSSSSS").getParser(),
-                DateTimeFormat.forPattern("yyyy-M-d H:m:s.SSSSSSSSS").getParser(),
-        };
-        @SuppressWarnings("SpellCheckingInspection")
-        DateTimePrinter timestampWithoutTimeZonePrinter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSSSSSSS").getPrinter();
-        HIVE_TIMESTAMP_PARSER = new DateTimeFormatterBuilder().append(timestampWithoutTimeZonePrinter, timestampWithoutTimeZoneParser).toFormatter().withZoneUTC();
-    }
+    private static final DateTimeFormatter HIVE_TIMESTAMP_PARSER = new DateTimeFormatterBuilder()
+            .appendPattern("yyyy-M-d[ H:m[:s]]")
+            .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+            .optionalStart().appendFraction(ChronoField.NANO_OF_SECOND, 1, 9, true).optionalEnd()
+            .toFormatter();
+    // TODO: switch to java.time when we implement writes with variable precision ()
```

Comment:
`()`?

why switch? (maybe remove the TODO)

Also, Joda may be more performant (used to be so)